controllers/front/ProductController.php

```diff
@@ -151,6 +151,9 @@ public function init()
                 header('HTTP/1.1 403 Forbidden');
                 header('Status: 403 Forbidden');
                 $this->errors[] = $this->trans('You do not have access to this product.', array(), 'Shop.Notifications.Error');
+                $this->setTemplate('errors/forbidden');
+
+                return;
```

Comment:
We are never sure, indeed. However, if another guy want to execute code for both cases, he will have to write it after this big `if { } else { }`. And your return will prevent this code to be executed.

Because having the `return;` statement doesn't change anything in the code executed here, I'm asking you to remove it.